test/util/memory.test.cpp

```diff
@@ -1,169 +0,0 @@
-#include <mbgl/test/stub_file_source.hpp>
-#include <mbgl/test/util.hpp>
-
-#include <mbgl/map/map.hpp>
-#include <mbgl/gl/headless_backend.hpp>
-#include <mbgl/gl/offscreen_view.hpp>
-#include <mbgl/util/default_thread_pool.hpp>
-#include <mbgl/util/io.hpp>
-#include <mbgl/util/run_loop.hpp>
-
-#include <algorithm>
-#include <iostream>
-#include <iterator>
-#include <string>
-#include <unordered_map>
-#include <utility>
-
-#include <stdlib.h>
-#include <unistd.h>
-
-using namespace mbgl;
-using namespace std::literals::string_literals;
-
-long getRSS() {
-    auto statm = util::read_file("/proc/self/statm");
-
-    std::vector<std::string> stats;
-    std::istringstream stream(statm);
-
-    std::copy(std::istream_iterator<std::string>(stream),
-        std::istream_iterator<std::string>(),
-        std::back_inserter(stats));
-
-    return std::stol(stats[1]) * getpagesize();
-}
-
-bool isUsingJemalloc() {
-    const char* preload = getenv("LD_PRELOAD");
-
-    if (preload) {
-        return std::string(preload).find("libjemalloc.so") != std::string::npos;
-    } else {
-        return false;
-    }
-}
-
-class MemoryTest {
-public:
-    MemoryTest() {
-        fileSource.styleResponse = [&](const Resource& res) { return response("style_" + getType(res) + ".json");};
-        fileSource.tileResponse = [&](const Resource& res) { return response(getType(res) + ".tile"); };
-        fileSource.sourceResponse = [&](const Resource& res) { return response("source_" + getType(res) + ".json"); };
-        fileSource.glyphsResponse = [&](const Resource&) { return response("glyphs.pbf"); };
-        fileSource.spriteJSONResponse = [&](const Resource&) { return response("sprite.json"); };
-        fileSource.spriteImageResponse = [&](const Resource&) { return response("sprite.png"); };
-    }
-
-    util::RunLoop runLoop;
-    HeadlessBackend backend { test::sharedDisplay() };
-    OffscreenView view{ backend.getContext(), { 512, 512 } };
-    StubFileSource fileSource;
-    ThreadPool threadPool { 4 };
-
-private:
-    Response response(const std::string& path) {
-        Response result;
-
-        auto it = cache.find(path);
-        if (it != cache.end()) {
-            result.data = it->second;
-        } else {
-            auto data = std::make_shared<std::string>(
-                util::read_file("test/fixtures/resources/"s + path));
-
-            cache.insert(it, std::make_pair(path, data));
-            result.data = data;
-        }
-
-        return result;
-    }
-
-    std::string getType(const Resource& res) {
-        if (res.url.find("satellite") != std::string::npos) {
-            return "raster";
-        } else {
-            return "vector";
-        }
-    };
-
-    std::unordered_map<std::string, std::shared_ptr<std::string>> cache;
-};
-
-TEST(Memory, Vector) {
```

Comment:
Do not remove this. These (Memory.Vector and Memory.Raster) are the only tests we have that do a full pass rendering inside valgrind looking for memory leaks.